coaddAnalysis.py

```diff
@@ -32,18 +46,38 @@ def getPatchConstituents(repo, band='g', printConstituents=False, verbose=False)
     constituentCountList : `list` of `int`
         The number of visits that contributed to each patch.
     """
-    butler = dafPersist.Butler(repo)
-    skymap = butler.get('deepCoadd_skyMap', dataId={'filter': band})
-    tractInfo = skymap[0]  # implicitly assumes there is just 1 tract
-    everyPatchList = [str(patch.getIndex()[0]) + ',' + str(patch.getIndex()[1]) for patch in tractInfo]
+    if gen == 'gen2':
+        butler = dafPersist.Butler(repo)
+        skymap = butler.get('deepCoadd_skyMap', dataId={'filter': band})
+        tract = skymap[0]  # implicitly assumes there is just 1 tract
+        everyPatchList = [str(patch.getIndex()[0]) + ',' + str(patch.getIndex()[1]) for patch in tract]
+    elif gen == 'gen3':
```

Comment:
In a previous function, you let gen3 be the default by using an `else` for cases like this. Would it make sense to do that here (and a few lines down)?

Reply:
Fixed several of these, good catch